src/QMCDrivers/MCPopulation.h

```diff
@@ -223,14 +230,25 @@ class MCPopulation
   const std::vector<RealType>& get_ptclgrp_mass() const { return ptclgrp_mass_; }
   const std::vector<RealType>& get_ptclgrp_inv_mass() const { return ptclgrp_inv_mass_; }
   const std::vector<RealType>& get_ptcl_inv_mass() const { return ptcl_inv_mass_; }
+  /// @}
 
-  /** }@ */
 
   /// check if all the internal vector contain consistent sizes;
   void checkIntegrity() const;
 
-  /// save walker configurations to walker_configs_ref_
+  /** save walker configurations to walker_configs_ref_
+   *  this is is just `R` and `spins` + possibly G, L
+   */
   void saveWalkerConfigurations(WalkerConfigurations& walker_configs);
+
+  /** Generator for walker_ids for this MCPopulation
+   *  This should be the single source for `walker_id`'s on this rank.
+   *  These are unique across MCPopulations with rank.
+   *  Defined as
+   *  \f$ walker_id = num_walkers_created_++ * num_ranks_ + rank_ + 1\f$
+   *  So starting from 1, 0 belongs to the golden walker.
```

Comment:
The formula of walker_id should be defined at the the top of MCpopulation. The documentation of a function needs to describe what a function does. It is hard to digest now.

The id of the gold walker is debatable. I tend to not include the golden walker at all. It doesn't participate calculation and should not. So it is not part of the population and doesn't deserve an ID. Even it is assigned an ID, the current formula is incorrect considering there is one golden walker per rank. `( 1+ num_walkers_created_++) * num_ranks_ + rank_ ` makes more sense.

It is acceptable but I don't feel comfortable that the proposed id formula produces gaps in ids.

In addition, it seems that the consumer of `nextWalkerID()` seems to know exactly what it does. It breaks encapsulation. Why is it public?

Reply:
It shouldn't be public, will change.

Reply:
The golden walker actually exists in MCPopulation only as a concept, it isn't actually present as a member and if it were it should be equivalent on all ranks so it doesn't need and or shouldn't have its own walker id.  It never moves so it is not logged.